### ISSUE TITLE
Put `Manager` inside application context

### ISSUE THREAD

**Issue by chshersh** [kowainik/hintman#69]

**Comment by vrom911** [comment]
Initial work is done here: https://github.com/kowainik/hintman/tree/vrom911/69-Put-Manager-inside-application-context
Seems like much more effort should be applied to get this done. Will get back to this one later.

**Comment by chshersh** [comment]
@vrom911 Makes sense. I checked your branch, looks nice :+1: And at the same time this indeed looks like too much work just for moving one variable to the environment :disappointed: Probably we can do this later, don't worry!
